In [28]:
import pandas as pd
import numpy as np

In [29]:
movie_industry = pd.read_csv("../data/movie_industry.csv", encoding = "ISO-8859-1" )
movie_industry.head()

budget                                 company country        director  \
0   8000000.0           Columbia Pictures Corporation     USA      Rob Reiner   
1   6000000.0                      Paramount Pictures     USA     John Hughes   
2  15000000.0                      Paramount Pictures     USA      Tony Scott   
3  18500000.0  Twentieth Century Fox Film Corporation     USA   James Cameron   
4   9000000.0                    Walt Disney Pictures     USA  Randal Kleiser   

       genre        gross                      name rating    released  \
0  Adventure   52287414.0               Stand by Me      R  1986-08-22   
1     Comedy   70136369.0  Ferris Bueller's Day Off  PG-13  1986-06-11   
2     Action  179800601.0                   Top Gun     PG  1986-05-16   
3     Action   85160248.0                    Aliens      R  1986-07-18   
4  Adventure   18564613.0   Flight of the Navigator     PG  1986-08-01   

   runtime  score               star   votes         writer  year  
0       89    8.1        Wil Wheaton  299174   Stephen King  1986  
1      103    7.8  Matthew Broderick  264740    John Hughes  1986  
2      110    6.9         Tom Cruise  236909       Jim Cash  1986  
3      137    8.4   Sigourney Weaver  540152  James Cameron  1986  
4       90    6.9        Joey Cramer   36636  Mark H. Baker  1986

In [30]:
movie_industry.loc[movie_industry.rating == "Not specified", "rating"] = "NOT RATED"
movie_industry.loc[movie_industry.rating == "UNRATED", "rating"] = "NOT RATED"

In [31]:
sum(movie_industry.gross.isna())

0

In [32]:
sum(movie_industry.gross == 0)

0

In [33]:
movie_industry.groupby('year').count()

budget  company  country  director  genre  gross  name  rating  \
year                                                                   
1986     220      220      220       220    220    220   220     220   
1987     219      219      219       219    219    219   219     219   
1988     220      220      220       220    220    220   220     220   
1989     221      221      221       221    221    221   221     221   
1990     220      220      220       220    220    220   220     220   
1991     220      220      220       220    220    220   220     220   
1992     220      220      220       220    220    220   220     220   
1993     220      220      220       220    220    220   220     220   
1994     220      220      220       220    220    220   220     220   
1995     220      220      220       220    220    220   220     220   
1996     220      220      220       220    220    220   220     220   
1997     220      220      220       220    220    220   220     220   
1998     220      220      220       220    220    220   220     220   
1999     220      220      220       220    220    220   220     220   
2000     220      220      220       220    220    220   220     220   
2001     220      220      220       220    220    220   220     220   
2002     220      220      220       220    220    220   220     220   
2003     220      220      220       220    220    220   220     220   
2004     220      220      220       220    220    220   220     220   
2005     220      220      220       220    220    220   220     220   
2006     220      220      220       220    220    220   220     220   
2007     220      220      220       220    220    220   220     220   
2008     220      220      220       220    220    220   220     220   
2009     220      220      220       220    220    220   220     220   
2010     220      220      220       220    220    220   220     220   
2011     220      220      220       220    220    220   220     220   
2012     220      220      220       220    220    220   220     220   
2013     220      220      220       220    220    220   220     220   
2014     220      220      220       220    220    220   220     220   
2015     220      220      220       220    220    220   220     220   
2016     220      220      220       220    220    220   220     220   

      released  runtime  score  star  votes  writer  
year                                                 
1986       220      220    220   220    220     220  
1987       219      219    219   219    219     219  
1988       220      220    220   220    220     220  
1989       221      221    221   221    221     221  
1990       220      220    220   220    220     220  
1991       220      220    220   220    220     220  
1992       220      220    220   220    220     220  
1993       220      220    220   220    220     220  
1994       220      220    220   220    220     220  
1995       220      220    220   220    220     220  
1996       220      220    220   220    220     220  
1997       220      220    220   220    220     220  
1998       220      220    220   220    220     220  
1999       220      220    220   220    220     220  
2000       220      220    220   220    220     220  
2001       220      220    220   220    220     220  
2002       220      220    220   220    220     220  
2003       220      220    220   220    220     220  
2004       220      220    220   220    220     220  
2005       220      220    220   220    220     220  
2006       220      220    220   220    220     220  
2007       220      220    220   220    220     220  
2008       220      220    220   220    220     220  
2009       220      220    220   220    220     220  
2010       220      220    220   220    220     220  
2011       220      220    220   220    220     220  
2012       220      220    220   220    220     220  
2013       220      220    220   220    220     220  
2014     

In [34]:
movie_industry[movie_industry.name.str.contains("Nemo")]

budget                  company country        director      genre  \
728   35000000.0        TMS Entertainment   Japan     Masami Hata     Family   
3750  94000000.0  Pixar Animation Studios     USA  Andrew Stanton  Animation   

            gross                                    name rating    released  \
728     1368210.0  Little Nemo: Adventures in Slumberland      G  1992-08-21   
3750  380843261.0                            Finding Nemo      G  2003-05-30   

      runtime  score           star   votes          writer  year  
728        85    7.2  Gabriel Damon    5800  Chris Columbus  1989  
3750      100    8.1  Albert Brooks  772029  Andrew Stanton  2003

In [35]:
years = np.unique(movie_industry.year.values)
years

array([1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996,
       1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007,
       2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016])

In [36]:
directors = np.unique(movie_industry.director.values)
directors

array(['A.R. Murugadoss', 'Aamir Khan', 'Aaron Blaise', ...,
       'Éric Rohmer', 'Étienne Chatiliez', 'Éva Gárdos'], dtype=object)

In [37]:
actors = np.unique(movie_industry.star.values)
actors

array(["'Weird Al' Yankovic", '50 Cent', 'A.J. Cook', ..., 'Álex Angulo',
       'Élodie Bouchez', 'Émilie Dequenne'], dtype=object)

In [38]:
individuals = np.append(directors, actors)
individuals

array(['A.R. Murugadoss', 'Aamir Khan', 'Aaron Blaise', ...,
       'Álex Angulo', 'Élodie Bouchez', 'Émilie Dequenne'], dtype=object)

In [39]:
years2 = np.repeat(years, len(individuals))
years2

array([1986, 1986, 1986, ..., 2016, 2016, 2016])

In [40]:
agents = np.tile(individuals, len(years))
agents

array(['A.R. Murugadoss', 'Aamir Khan', 'Aaron Blaise', ...,
       'Álex Angulo', 'Élodie Bouchez', 'Émilie Dequenne'], dtype=object)

In [41]:
is_director = np.tile(np.append(np.ones(len(directors)), np.zeros(len(actors))), len(years))
is_director

array([1., 1., 1., ..., 0., 0., 0.])

In [42]:
ID = np.tile(np.arange(len(individuals)), len(years))
ID

array([   0,    1,    2, ..., 5260, 5261, 5262])

In [43]:
time = np.repeat(np.arange(len(years)), len(individuals))
time

array([ 0,  0,  0, ..., 30, 30, 30])

In [44]:
assert len(agents) == len(years2)
assert len(agents) == len(is_director)
assert len(agents) == len(ID)

In [45]:
df  = pd.DataFrame({"agent": agents,
                   "is_director": is_director,
                   "id": ID,
                   "year": years2,
                   "time": time})
df

agent  is_director    id  year  time
0          A.R. Murugadoss          1.0     0  1986     0
1               Aamir Khan          1.0     1  1986     0
2             Aaron Blaise          1.0     2  1986     0
3             Aaron Harvey          1.0     3  1986     0
4       Aaron Kim Johnston          1.0     4  1986     0
...                    ...          ...   ...   ...   ...
163148         Zoey Deutch          0.0  5258  2016    30
163149     Zooey Deschanel          0.0  5259  2016    30
163150         Álex Angulo          0.0  5260  2016    30
163151      Élodie Bouchez          0.0  5261  2016    30
163152     Émilie Dequenne          0.0  5262  2016    30

[163153 rows x 5 columns]

In [46]:
assert df.groupby("is_director").count().agent[1] == len(directors)*len(years)
assert len(df.groupby(["agent", "is_director"]).count()) == len(individuals)

In [47]:
len(individuals)  - len(df.groupby(["agent"]).count())

171

There are 171 people who double and actors and directors

In [48]:
sum(el in directors for el in actors)

171

In [49]:
%%time
total_gross = []
mean_score = []
mean_votes = []
for i in df.index:
    if df.is_director[i]:
        temp = movie_industry[movie_industry.director == df.agent[i]]
        temp = temp[temp.year <= df.year[i]]
        if len(temp) > 0 :  
            gross = sum(temp.gross.values)
            score = np.mean(temp.score.values)
            votes = np.mean(temp.votes.values)
        else:
            gross = 0
            score = 0
            votes = 0
        total_gross.append(gross)
        mean_score.append(score)
        mean_votes.append(votes)
    else:
        temp = movie_industry[movie_industry.star == df.agent[i]]
        temp = temp[temp.year <= df.year[i]]
        if len(temp) > 0 :  
            gross = sum(temp.gross.values)
            score = np.mean(temp.score.values)
            votes = np.mean(temp.votes.values)
        else:
            gross = 0
            score = 0
            votes = 0
        total_gross.append(gross)
        mean_score.append(score)
        mean_votes.append(votes)

CPU times: user 5min 10s, sys: 5.34 s, total: 5min 16s
Wall time: 5min 42s


In [52]:
df["log_total_gross"] = [np.log(i+1) for i in total_gross]
df["mean_score"] = mean_score
df["mean_votes"] = mean_votes

In [53]:
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637241   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729861   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474098   

        mean_score  mean_votes  
0             0.00         0.0  
1             0.00         0.0  
2             0.00         0.0  
3             0.00         0.0  
4             0.00         0.0  
...            ...         ...  
163148        5.95     54405.5  
163149        7.30    208674.5  
163150        7.50     13867.0  
163151        7.60      4397.0  
163152        7.60     10176.0  

[163153 rows x 8 columns]

In [54]:
movie_industry.groupby("genre").count().sort_values("director", ascending= False)

budget  company  country  director  gross  name  rating  released  \
genre                                                                          
Comedy       2080     2080     2080      2080   2080  2080    2080      2080   
Drama        1444     1444     1444      1444   1444  1444    1444      1444   
Action       1331     1331     1331      1331   1331  1331    1331      1331   
Crime         522      522      522       522    522   522     522       522   
Adventure     392      392      392       392    392   392     392       392   
Biography     359      359      359       359    359   359     359       359   
Horror        277      277      277       277    277   277     277       277   
Animation     277      277      277       277    277   277     277       277   
Mystery        38       38       38        38     38    38      38        38   
Fantasy        32       32       32        32     32    32      32        32   
Thriller       18       18       18        18     18    18      18        18   
Romance        15       15       15        15     15    15      15        15   
Family         14       14       14        14     14    14      14        14   
Sci-Fi         13       13       13        13     13    13      13        13   
Musical         4        4        4         4      4     4       4         4   
War             2        2        2         2      2     2       2         2   
Western         2        2        2         2      2     2       2         2   

           runtime  score  star  votes  writer  year  
genre                                                 
Comedy        2080   2080  2080   2080    2080  2080  
Drama         1444   1444  1444   1444    1444  1444  
Action        1331   1331  1331   1331    1331  1331  
Crime          522    522   522    522     522   522  
Adventure      392    392   392    392     392   392  
Biography      359    359   359    359     359   359  
Horror         277    277   277    277     277   277  
Animation      277    277   277    277     277   277  
Mystery         38     38    38     38      38    38  
Fantasy         32     32    32     32      32    32  
Thriller        18     18    18     18      18    18  
Romance         15     15    15     15      15    15  
Family          14     14    14     14      14    14  
Sci-Fi          13     13    13     13      13    13  
Musical          4      4     4      4       4     4  
War              2      2     2      2       2     2  
Western          2      2     2      2       2     2

In [55]:
genre = movie_industry.groupby("genre").count().sort_values("budget", ascending= False).index.values[0:8]
genre

array(['Comedy', 'Drama', 'Action', 'Crime', 'Adventure', 'Biography',
       'Horror', 'Animation'], dtype=object)

In [56]:
%%time
d = {}
for gen in genre:
    d[gen] = []   
for i in df.index:
    if df.is_director[i]:
        temp = movie_industry[movie_industry.director == df.agent[i]]
    else:
        temp = movie_industry[movie_industry.star == df.agent[i]]
    temp = temp[temp.year <= df.year[i]]
    if len(temp) > 0:
        temp = temp.groupby(["genre"])["director"].count()
        for gen in genre:
            if gen in temp.index:
                d[gen].append(temp[gen])
            else:
                d[gen].append(0)
    else:
        for gen in genre:
            d[gen].append(0)

CPU times: user 6min 45s, sys: 7.95 s, total: 6min 53s
Wall time: 7min 28s


In [58]:
temp = pd.DataFrame(d)
df = df.merge(temp, left_index =True, right_index =True)
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637241   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729861   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474098   

        mean_score  mean_votes  Comedy  Drama  Action  Crime  Adventure  \
0             0.00         0.0       0      0       0      0          0   
1             0.00         0.0       0      0       0      0          0   
2             0.00         0.0       0      0       0      0          0   
3             0.00         0.0       0      0       0      0          0   
4             0.00         0.0       0      0       0      0          0   
...            ...         ...     ...    ...     ...    ...        ...   
163148        5.95     54405.5       1      0       1      0          0   
163149        7.30    208674.5       1      1       0      0          0   
163150        7.50     13867.0       0      0       1      0          0   
163151        7.60      4397.0       0      1       0      0          0   
163152        7.60     10176.0       0      1       0      0          0   

        Biography  Horror  Animation  
0               0       0          0  
1               0       0          0  
2               0       0          0  
3               0       0          0  
4               0       0          0  
...           ...     ...        ...  
163148          0       0          0  
163149          0       0          0  
163150          0       0          0  
163151          0       0          0  
163152          0       0          0  

[163153 rows x 16 columns]

In [59]:
movie_industry.groupby("rating").count().sort_values("director", ascending= False)

budget  company  country  director  genre  gross  name  released  \
rating                                                                        
R            3392     3392     3392      3392   3392   3392  3392      3392   
PG-13        1995     1995     1995      1995   1995   1995  1995      1995   
PG            951      951      951       951    951    951   951       951   
NOT RATED     308      308      308       308    308    308   308       308   
G             147      147      147       147    147    147   147       147   
NC-17          22       22       22        22     22     22    22        22   
B               1        1        1         1      1      1     1         1   
B15             1        1        1         1      1      1     1         1   
TV-14           1        1        1         1      1      1     1         1   
TV-MA           1        1        1         1      1      1     1         1   
TV-PG           1        1        1         1      1      1     1         1   

           runtime  score  star  votes  writer  year  
rating                                                
R             3392   3392  3392   3392    3392  3392  
PG-13         1995   1995  1995   1995    1995  1995  
PG             951    951   951    951     951   951  
NOT RATED      308    308   308    308     308   308  
G              147    147   147    147     147   147  
NC-17           22     22    22     22      22    22  
B                1      1     1      1       1     1  
B15              1      1     1      1       1     1  
TV-14            1      1     1      1       1     1  
TV-MA            1      1     1      1       1     1  
TV-PG            1      1     1      1       1     1

In [60]:
ratings = movie_industry.groupby("rating").count().sort_values("director", ascending= False).index.values[0:5]
ratings

array(['R', 'PG-13', 'PG', 'NOT RATED', 'G'], dtype=object)

In [61]:
%%time
d = {}
for gen in ratings:
    d[gen] = []   
for i in df.index:
    if df.is_director[i]:
        temp = movie_industry[movie_industry.director == df.agent[i]]
    else:
        temp = movie_industry[movie_industry.star == df.agent[i]]
    temp = temp[temp.year <= df.year[i]]
    if len(temp) > 0:
        temp = temp.groupby(["rating"])["director"].count()
        for gen in ratings:
            if gen in temp.index:
                d[gen].append(temp[gen])
            else:
                d[gen].append(0)
    else:
        for gen in ratings:
            d[gen].append(0)

CPU times: user 6min 1s, sys: 5.91 s, total: 6min 7s
Wall time: 6min 28s


In [64]:
temp = pd.DataFrame(d)
df = df.merge(temp, left_index =True, right_index =True)
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637241   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729861   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474098   

        mean_score  mean_votes  Comedy  Drama  ...  Crime  Adventure  \
0             0.00         0.0       0      0  ...      0          0   
1             0.00         0.0       0      0  ...      0          0   
2             0.00         0.0       0      0  ...      0          0   
3             0.00         0.0       0      0  ...      0          0   
4             0.00         0.0       0      0  ...      0          0   
...            ...         ...     ...    ...  ...    ...        ...   
163148        5.95     54405.5       1      0  ...      0          0   
163149        7.30    208674.5       1      1  ...      0          0   
163150        7.50     13867.0       0      0  ...      0          0   
163151        7.60      4397.0       0      1  ...      0          0   
163152        7.60     10176.0       0      1  ...      0          0   

        Biography  Horror  Animation  R  PG-13  PG  NOT RATED  G  
0               0       0          0  0      0   0          0  0  
1               0       0          0  0      0   0          0  0  
2               0       0          0  0      0   0          0  0  
3               0       0          0  0      0   0          0  0  
4               0       0          0  0      0   0          0  0  
...           ...     ...        ... ..    ...  ..        ... ..  
163148          0       0          0  1      1   0          0  0  
163149          0       0          0  1      1   0          0  0  
163150          0       0          0  1      0   0          0  0  
163151          0       0          0  0      0   0          1  0  
163152          0       0          0  1      0   0          0  0  

[163153 rows x 21 columns]

In [65]:
oscar = pd.read_csv("../data/the_oscar_award.csv")
oscar.category = oscar.category.str.lower()
filtered = oscar[oscar.category.str.contains("(actor)|(actress)|(directing)", regex = True)]
filtered.loc[filtered.category.str.contains("(actor)|(actress)"), 'is_director'] = 0
filtered.loc[filtered.category.str.contains("directing"), 'is_director'] = 1

/Users/shashank/opt/anaconda3/lib/python3.7/site-packages/pandas/core/strings.py:1843: UserWarning: This pattern has match groups. To actually get the groups, use str.extract.
  return func(self, *args, **kwargs)
/Users/shashank/opt/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:376: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj[key] = _infer_fill_value(value)
/Users/shashank/opt/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py:494: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-c

In [66]:
oscar_ppl = filtered[~filtered.name.str.contains("( and)|(,)|(and )", regex = True)]
oscar_ppl.is_director = oscar_ppl.is_director.astype(np.uint8)
oscar_ppl.head()

/Users/shashank/opt/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py:5208: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


year_film  year_ceremony  ceremony category                 name  \
0       1927           1928         1    actor  Richard Barthelmess   
1       1927           1928         1    actor        Emil Jannings   
2       1927           1928         1  actress       Louise Dresser   
3       1927           1928         1  actress         Janet Gaynor   
4       1927           1928         1  actress       Gloria Swanson   

               film  winner  is_director  
0         The Noose   False            0  
1  The Last Command    True            0  
2   A Ship Comes In   False            0  
3        7th Heaven    True            0  
4    Sadie Thompson   False            0

In [67]:
oscar_ppl.shape

(2192, 8)

In [68]:
filtered[filtered.name.str.contains("( and)|(,)|(and )", regex = True)]

year_film  year_ceremony  ceremony                    category  \
531        1937           1938        10  actor in a supporting role   
3768       1961           1962        34                   directing   
5655       1978           1979        51                   directing   
5942       1981           1982        54  actor in a supporting role   
6043       1982           1983        55  actor in a supporting role   
6289       1984           1985        57                   directing   
6506       1986           1987        59                   directing   
7558       1996           1997        69  actor in a supporting role   
8832       2007           2008        80                   directing   
9184       2010           2011        83                   directing   

                             name                        film  winner  \
531                  Roland Young                      Topper   False   
3768  Robert Wise, Jerome Robbins             West Side Story    True   
5655    Warren Beatty, Buck Henry             Heaven Can Wait   False   
5942       Howard E. Rollins, Jr.                     Ragtime   False   
6043           Louis Gossett, Jr.  An Officer and a Gentleman    True   
6289                 Roland Joffé          The Killing Fields   False   
6506                 Roland Joffé                 The Mission   False   
7558            Cuba Gooding, Jr.               Jerry Maguire    True   
8832     Joel Coen and Ethan Coen      No Country for Old Men    True   
9184     Joel Coen and Ethan Coen                   True Grit   False   

      is_director  
531           0.0  
3768          1.0  
5655          1.0  
5942          0.0  
6043          0.0  
6289          1.0  
6506          1.0  
7558          0.0  
8832          1.0  
9184          1.0

In [69]:
#Manually modify the rows where there are special characters (for movie after 1986)
oscar_ppl = pd.concat([oscar_ppl, filtered[(filtered.index==7558) | (filtered.index==6506)]])
oscar_ppl.shape

(2194, 8)

In [70]:
oscar_ppl[oscar_ppl.name=="Joel Coen and Ethan Coen"]

Empty DataFrame
Columns: [year_film, year_ceremony, ceremony, category, name, film, winner, is_director]
Index: []

In [71]:
oscar_ppl = pd.concat([oscar_ppl, filtered[(filtered.index==8832) | (filtered.index==9184)]], ignore_index=False)
oscar_ppl = pd.concat([oscar_ppl, filtered[(filtered.index==8832) | (filtered.index==9184)]], ignore_index=True)
oscar_ppl.shape

(2198, 8)

In [72]:
oscar_ppl[oscar_ppl.name== "Joel Coen and Ethan Coen"]

year_film  year_ceremony  ceremony   category                      name  \
2194       2007           2008        80  directing  Joel Coen and Ethan Coen   
2195       2010           2011        83  directing  Joel Coen and Ethan Coen   
2196       2007           2008        80  directing  Joel Coen and Ethan Coen   
2197       2010           2011        83  directing  Joel Coen and Ethan Coen   

                        film  winner  is_director  
2194  No Country for Old Men    True          1.0  
2195               True Grit   False          1.0  
2196  No Country for Old Men    True          1.0  
2197               True Grit   False          1.0

In [73]:
oscar_ppl.at[2194, 'name'] = 'Joel Coen'
oscar_ppl.at[2195, 'name'] = 'Ethan Coen'
oscar_ppl.at[2196, 'name'] = 'Ethan Coen'
oscar_ppl.at[2197, 'name'] = 'Joel Coen'

In [74]:
oscar_ppl.shape

(2198, 8)

In [75]:
oscar_ppl[oscar_ppl.name=='Alejandro G. Iñárritu'].index
oscar_ppl.at[2062, 'name'] = 'Alejandro González Iñárritu'

In [76]:
oscar_ppl = oscar_ppl[oscar_ppl['year_film']>1985]
print('oscar award dataset has shape {}'.format(oscar_ppl.shape))
oscar_ppl.head()

oscar award dataset has shape (852, 8)


year_film  year_ceremony  ceremony                 category  \
1346       1986           1987        59  actor in a leading role   
1347       1986           1987        59  actor in a leading role   
1348       1986           1987        59  actor in a leading role   
1349       1986           1987        59  actor in a leading role   
1350       1986           1987        59  actor in a leading role   

               name                      film  winner  is_director  
1346  Dexter Gordon           'Round Midnight   False          0.0  
1347    Bob Hoskins                 Mona Lisa   False          0.0  
1348   William Hurt  Children of a Lesser God   False          0.0  
1349    Paul Newman        The Color of Money    True          0.0  
1350    James Woods                  Salvador   False          0.0

In [77]:
%%time
total_nominations =[]
total_win_counts=[]
for i in df.index:
    if df.is_director[i]:
        temp = oscar_ppl[(oscar_ppl.name == df.agent[i]) & (oscar_ppl.is_director == True)]
        temp = temp[temp.year_film <= df.year[i]]
        nomination = temp.shape[0]
        total_nominations.append(nomination)
        win = sum(temp.winner.values)
        total_win_counts.append(win)
    else:
        temp = oscar_ppl[(oscar_ppl.name == df.agent[i]) & (oscar_ppl.is_director == False)]
        temp = temp[temp.year_film <= df.year[i]]
        nomination = temp.shape[0]
        total_nominations.append(nomination)
        win = sum(temp.winner.values)
        total_win_counts.append(win)

CPU times: user 6min 23s, sys: 6.65 s, total: 6min 30s
Wall time: 6min 58s


In [78]:
df["nominations"] = total_nominations
df["win_counts"] = total_win_counts
df.head()

agent  is_director  id  year  time  log_total_gross  \
0     A.R. Murugadoss          1.0   0  1986     0              0.0   
1          Aamir Khan          1.0   1  1986     0              0.0   
2        Aaron Blaise          1.0   2  1986     0              0.0   
3        Aaron Harvey          1.0   3  1986     0              0.0   
4  Aaron Kim Johnston          1.0   4  1986     0              0.0   

   mean_score  mean_votes  Comedy  Drama  ...  Biography  Horror  Animation  \
0         0.0         0.0       0      0  ...          0       0          0   
1         0.0         0.0       0      0  ...          0       0          0   
2         0.0         0.0       0      0  ...          0       0          0   
3         0.0         0.0       0      0  ...          0       0          0   
4         0.0         0.0       0      0  ...          0       0          0   

   R  PG-13  PG  NOT RATED  G  nominations  win_counts  
0  0      0   0          0  0            0           0  
1  0      0   0          0  0            0           0  
2  0      0   0          0  0            0           0  
3  0      0   0          0  0            0           0  
4  0      0   0          0  0            0           0  

[5 rows x 23 columns]

In [79]:
df

agent  is_director    id  year  time  log_total_gross  \
0          A.R. Murugadoss          1.0     0  1986     0         0.000000   
1               Aamir Khan          1.0     1  1986     0         0.000000   
2             Aaron Blaise          1.0     2  1986     0         0.000000   
3             Aaron Harvey          1.0     3  1986     0         0.000000   
4       Aaron Kim Johnston          1.0     4  1986     0         0.000000   
...                    ...          ...   ...   ...   ...              ...   
163148         Zoey Deutch          0.0  5258  2016    30        18.035911   
163149     Zooey Deschanel          0.0  5259  2016    30        17.310201   
163150         Álex Angulo          0.0  5260  2016    30         9.637241   
163151      Élodie Bouchez          0.0  5261  2016    30        13.729861   
163152     Émilie Dequenne          0.0  5262  2016    30        12.474098   

        mean_score  mean_votes  Comedy  Drama  ...  Biography  Horror  \
0             0.00         0.0       0      0  ...          0       0   
1             0.00         0.0       0      0  ...          0       0   
2             0.00         0.0       0      0  ...          0       0   
3             0.00         0.0       0      0  ...          0       0   
4             0.00         0.0       0      0  ...          0       0   
...            ...         ...     ...    ...  ...        ...     ...   
163148        5.95     54405.5       1      0  ...          0       0   
163149        7.30    208674.5       1      1  ...          0       0   
163150        7.50     13867.0       0      0  ...          0       0   
163151        7.60      4397.0       0      1  ...          0       0   
163152        7.60     10176.0       0      1  ...          0       0   

        Animation  R  PG-13  PG  NOT RATED  G  nominations  win_counts  
0               0  0      0   0          0  0            0           0  
1               0  0      0   0          0  0            0           0  
2               0  0      0   0          0  0            0           0  
3               0  0      0   0          0  0            0           0  
4               0  0      0   0          0  0            0           0  
...           ... ..    ...  ..        ... ..          ...         ...  
163148          0  1      1   0          0  0            0           0  
163149          0  1      1   0          0  0            0           0  
163150          0  1      0   0          0  0            0           0  
163151          0  0      0   0          1  0            0           0  
163152          0  1      0   0          0  0            0           0  

[163153 rows x 23 columns]

In [80]:
df.to_csv("../data/features.csv", index= False)